File: exercism/go/rna-transcription/rna_transcription.go
// Package strand contains a function that performs RNA transcription.
package strand

var rnaTable = map[rune]rune {
	'G': 'C',
	'C': 'G',
	'T': 'A',
	'A': 'U',
}

// ToRNA transcripts a DNA sequence to its complement RNA sequence.
func ToRNA(dna string) string {
	var result []rune
	for _, d := range dna {
		r, ok := rnaTable[d]
		if !ok {
			panic("unexpected input nucleotide")
		}
		result = append(result, r)
	}
	return string(result)
}

File: exercism/clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(defn to-rna [dna]
  (clojure.string/escape
   dna
   (fn [ch]
     (case ch
       \C \G
       \G \C
       \T \A
       \A \U
       (assert false :invalid-char)))))
